### PR TITLE
Add VB LanguageVersion 15.6

### PR DESCRIPTION
**Pull request by jcouv**
In preparation for dev15.6 work. Corresponding change for C#: https://github.com/dotnet/roslyn/issues/19846.

@dotnet/roslyn-compiler for review. Thanks

### CODE REVIEW

**Comment by AlekseyTs** [comment]
Are we planning any language changes for VB in 15.6? if not, we probably shouldn't add "an empty" version.

File: src/Compilers/VisualBasic/Portable/PredefinedPreprocessorSymbols.vb [hunk 1]
```diff
@@ -1,14 +1,11 @@
...
 Imports System.Collections.Immutable
-Imports Microsoft.CodeAnalysis.Text
-Imports Microsoft.CodeAnalysis.VisualBasic.Symbols
-Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
 
 Namespace Microsoft.CodeAnalysis.VisualBasic
     Public Module PredefinedPreprocessorSymbols
 
-        Friend Const CurrentVersionNumber = 15.3
+        Friend Const CurrentVersionNumber = 15.6
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/PredefinedPreprocessorSymbols.vb]: > Friend Const CurrentVersionNumber = 15.6 [](start = 7, length = 41)

This looks very fragile, it is pretty easy to miss this change when a new version is added. I suggest we change this constant to a readonly shared field with initializer that calls MapSpecifiedToEffectiveVersion with latest, then calls GetErrorName on the result and parses its return as Double.

Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/PredefinedPreprocessorSymbols.vb]: >Friend Const CurrentVersionNumber = 15.6 [](start = 8, length = 40)

Do we have a test covering this change?

**Comment by jaredpar** [comment]
> Are we planning any language changes for VB in 15.6? if not, we probably shouldn't add "an empty" version.

There are a number of C# constructs that will require work to recognize in VB: `ref readonly` and `ref like` types.  Lacking an VB changes they would be cut out of consuming the C# APIs due to mod reqs.

**Comment by jcouv** [comment]
@AlekseyTs Yes, I'm confident that VB 15.6 will have non-trailing named arguments. 
Since I wasn't sure if there would be some other feature, I thought it's better to make this change in common branch than feature branch. But I'm ok to cherry-pick into feature branch if unsure.

**Comment by AlekseyTs** [comment]
Done with review pass.

**Comment by AlekseyTs** [comment]
Regarding whether we are going to take advantage of the 15.6 version, I think it should be fine to open an issue to remove the version if it is unused at the end.

File: src/Compilers/VisualBasic/Portable/PredefinedPreprocessorSymbols.vb (hunk 1 above)
Comment (jcouv) [inline src/Compilers/VisualBasic/Portable/PredefinedPreprocessorSymbols.vb]: Yes, a test actually caught this (I had forgotten).

Comment (jcouv) [inline src/Compilers/VisualBasic/Portable/PredefinedPreprocessorSymbols.vb]: Here's the specific test: `Microsoft.CodeAnalysis.VisualBasic.UnitTests.VisualBasicParseOptionsTests.CurrentVersionNumber`

**Comment by jcouv** [comment]
Filed https://github.com/dotnet/roslyn/issues/19903 in case it ends up un-used.

File: src/Compilers/VisualBasic/Portable/PredefinedPreprocessorSymbols.vb (hunk 1 above)
Comment (jcouv) [inline src/Compilers/VisualBasic/Portable/PredefinedPreprocessorSymbols.vb]: Will do. Thanks for the specific proposal.

**Pull request merged**